### ISSUE TITLE
Account Bootstrap/Update

### ISSUE THREAD

**Issue by veselosky** [veselosky/bluebucket#6]
As a (potential) user of Blue Bucket
I want a browser-based tool that will help me set up the system in my own AWS account
So that I can get started with the system without being a techie.

In this version of the application, we will keep it simple to keep it fast. You will need to have AWS credentials from your account, and feed them to the browser app for it to do its job. In a future version, we may introduce multi-tenancy with a separate and easier authentication system.

### Tasks

- [ ] Create a form that gathers AWS credentials. When the form is submitted, the credentials will be stored in local storage, and will be used to configure the AWS client library. On load, check for credentials in local storage. If found, configure the AWS library and skip the credentials form.
- [ ] Check the AWS account for existing Blue Bucket Updater Lambda function. If found, invoke the Lambda function in query mode to get a list of updates available. Present an offer to install updates. If the Updater Lambda is not found, offer to install:
    * Assure: a Role to be assumed by Lambda functions.
        * Grant full permission over Lambda services
        * Grant full permission over S3 services
        * Grant full permission over API Gateway services (for future)
    * if necessary, Install latest Lambda Blue Bucket Updater
    * invoke the Updater Lambda to complete the installation/updates.

Once all is good, forward the user to the Bucket Management Form.

**Comment by aarcro** [comment]
That sounds kinda scary. Can you document for the user how to create an IAM account with just the required privs?

**Comment by veselosky** [comment]
Yeah, this is just for the MVP so I can work with the system myself. (Version 0.1 is the "get something working ugly" edition.) When it gets close to being usable by humans, I'll document the AWS setup clearly for normal people. Since the secret is only stored in local storage and never actually sent anywhere, this seems reasonably safe, even if they user their root credentials. As long as we guard against XSS attacks, anyway.

**Comment by aarcro** [comment]
Might want to take a look at https://www.terraform.io/intro/use-cases.html terraform for automating buildout.
